src/optimise/optimisers.jl

```diff
@@ -23,7 +24,8 @@ function apply!(o::Descent, x, Δ)
 end
 
 """
-    Momentum(η = 0.01; ρ = 0.9)
+    Momentum() => Momentum(η, ρ)
```

Comment:
Making clear that the parameters aren’t keywords is great, but these `=>` things seem very odd to me. Isn’t it enought to write the defaults next to or below the definition? 
```suggestion
    Momentum(η, ρ) # Defaults: η = 0.01, ρ = 0.9
```

Reply:
I think we also need to make clear that users don't need to specify the parameters if they want to use the defaults. That was kind of the hope with the previous docstring since Julia doesn't do named positional arguments, and defining defaults values like that is pretty common practice.

I'd be happy to make this change if there is consensus.

Reply:
Agree with the goals, I just think `=>` is confusing. Maybe another way is to write this: 
```
    Momentum(η, ρ)
    Momentum() # Defaults: η = 0.01, ρ = 0.9
```

Reply:
Right, I think the last commit is close to this, save the `$opt()`

Maybe we add that in

Reply:
Maybe we can just use `=`. That's exactly how you'd define it in Julia anyway, so it should be unambiguous.

Reply:
If we use something like `Defaults`, it should definitely be in the text rather than the signature; perhaps with a description of the parameters. That might actually be better since we can expand eta to "learning rate" etc. Probably it's a good idea to do both things.